pkg/monitoring/mimir/service.go

```diff
@@ -60,6 +61,30 @@ func (ms *MimirService) CreateApiKey(ctx context.Context, logger logr.Logger) er
 	current := &corev1.Secret{}
 	err := ms.Client.Get(ctx, objectKey, current)
 	if apierrors.IsNotFound(err) {
+		// First all secrets using the password from the mimirApiKey secret are deleted
+		// to ensure that they won't use an outdated password.
+		logger.Info("Deleting old secrets")
```

Comment:
How do you make sure the secret is recreated before the next reconciliation?

Reply:
Well, since this deletion is happening in the case where the basic-auth secret isn't present on the cluster and considering that the remotewrite secrets and ingress secret all need the basic-auth secret's data and can't be created without it, as long as this process is not finished and the basic-auth created, they won't be created before.

Not sure if I explained it clearly :sweat_smile:

Reply:
alright :)